src/tests/Microsoft.PowerFx.Interpreter.Tests.Shared/RecalcEngineTests.cs

```diff
@@ -793,30 +793,32 @@ public void DelegableUDFTest()
 
             var recalcEngine = new RecalcEngine(config);
 
-            recalcEngine.AddUserDefinedFunction("A():MyDataSourceTableType = Filter(MyDataSource, Value > 10);C():MyDataSourceTableType = A(); B():MyDataSourceTableType = Filter(C(), Value > 11); D():MyDataSourceTableType = { Filter(B(), Value > 12); };", CultureInfo.InvariantCulture, symbolTable: recalcEngine.EngineSymbols, allowSideEffects: true);
-            var func = recalcEngine.Functions.WithName("A").First() as UserDefinedFunction;
+            recalcEngine.AddUserDefinedFunction("A():MyDataSourceTableType = Filter(MyDataSource, Value > 10);C():MyDataSourceTableType = A(); B():MyDataSourceTableType = Filter(C(), Value > 11); D():MyDataSourceTableType = { Filter(B(), Value > 12); }; F():MyDataSourceTableType = MyDataSource;", CultureInfo.InvariantCulture, symbolTable: recalcEngine.EngineSymbols, allowSideEffects: true);
 
+            var func = recalcEngine.Functions.WithName("A").First() as UserDefinedFunction;
             Assert.True(func.IsAsync);
             Assert.True(func.IsDelegatable);
 
             func = recalcEngine.Functions.WithName("C").First() as UserDefinedFunction;
-
             Assert.True(func.IsAsync);
             Assert.True(func.IsDelegatable);
 
             func = recalcEngine.Functions.WithName("B").First() as UserDefinedFunction;
-
             Assert.True(func.IsAsync);
             Assert.True(func.IsDelegatable);
 
             func = recalcEngine.Functions.WithName("D").First() as UserDefinedFunction;
-
+            
             // Imperative function is not delegable
             Assert.True(func.IsAsync);
             Assert.True(!func.IsDelegatable);
 
             // Binding fails for recursive definitions and hence function is not added.
             Assert.False(recalcEngine.AddUserDefinedFunction("E():Void = { E(); };", CultureInfo.InvariantCulture, symbolTable: recalcEngine.EngineSymbols, allowSideEffects: true).IsSuccess);
+            
+            func = recalcEngine.Functions.WithName("F").First() as UserDefinedFunction;
+            Assert.True(func.IsAsync);
+            Assert.True(func.IsDelegatable);

```

Comment:
Can you add a testcase where IsDelegatable is false?

Reply:
For example,
 Filter(MyDataSource, Value < 10)

Reply:
This seems to be another bug. It returns IsDelegatable = true

Reply:
https://github.com/microsoft/Power-Fx/blame/0fa19687742caa771c4a8f0716f07005497bec49/src/libraries/Microsoft.PowerFx.Core/Functions/Delegation/DelegationStrategies/OpDelegationStrategy.cs#L281-L287C14

This is by design not checking on FirstNameNode.

Reply:
Added a mock non delegatable data source for unit tests